internal/sliceutils/sliceutils.go

```diff
@@ -47,3 +50,25 @@ func intsContain(ns []int, v int) bool {
 func InsertStringAt(data []string, i int, v string) []string {
 	return append(data[:i], append([]string{v}, data[i:]...)...)
 }
+
+func Push[T any](data []T, v T) []T {
+	return append(data, v)
+}
+
+func Pop[T any](data []T) (T, []T, error) {
+	var t T
+	if len(data) == 0 {
+		return t, nil, errs.New("Cannot pop from empty slice")
+	}
+
+	return data[len(data)-1], data[:len(data)-1], nil
```

Comment:
Nit: Seeing as the slice is modified in memory there doesn't seem to be any utility in having it return the slice (second arg).

Reply:
I believe that because the size of the underlying array is changing the slice isn't modified outside of the function. I was messing around trying to get it to work but it would require passing a pointer to the slice.

Reply:
huh, interesting.. I wasn't aware of that behaviour.